syft/frameworks/torch/tensors/interpreters/chinese_remainder_theorem.py

```diff
@@ -0,0 +1,406 @@
+        """
+        assert field is None or field == self.field, "field is chosen when fixing precision"
+        for mod, res in self.child.items():
+            ast_mod = res.share(*owners, field=mod, crypto_provider=None)
+            self.child[mod] = ast_mod
+
+        return self
+
+    def get(self):
+        """ Get back a tensor shared between several workers.
+        """
+        self_ = self.child
+        for mod, res in self_.items():
+            res.child = res.child.get().child
+        return self.wrap()
+
+    @staticmethod
+    def simplify(tensor: "CRTTensor") -> tuple:
+        """
+        This function takes the attributes of a CRTTensor and saves them in a tuple
+        Args:
+            crttensor: a CRTTensor
+        Returns:
+            tuple: a tuple holding the unique attributes of the tensor
+        """
+        chain = None
+        if hasattr(tensor, "child"):
+            chain = syft.serde._simplify(tensor.child)
+
+        return (
+            tensor.id,
+            tensor.base,
+            tensor.precision_fractional,
+            # tensor.reconstruction_coeffs,
+            chain,
+        )
+
+    @staticmethod
+    def detail(worker, tensor_tuple: tuple) -> "CRTTensor":
+        """
+        This function reconstructs a CRTTensor given its attributes in form of a tuple.
+        Args:
+            worker: the worker doing the deserialization
+            tensor_tuple: a tuple holding the attributes of the CRTTensor
+        Returns:
+            CRTTensor: a CRTTensor
+        """
+        tensor_id, tensor_base, tensor_precision_fractional, chain = tensor_tuple
+
+        tensor = syft.CRTTensor(
+            base=tensor_base,
+            precision_fractional=tensor_precision_fractional,
+            # FIXME need to cast to tuple otherwise tensor_reconstruction_coeffs is an unashable list
+            # reconstruction_coeffs can be large integers that cannot be serialized with msgpack
+            # reconstruction_coeffs=tuple(tensor_reconstruction_coeffs),
+            owner=worker,
+            id=tensor_id,
+        )
+
+        if chain is not None:
+            chain = syft.serde._detail(worker, chain)
+            tensor.child = chain
+
+        # We need it now because we cannot serialize large reconstruction coefficients
+        tensor.reconstruction_coeffs = tensor.compute_reconstruction_coeffs()
+
+        return tensor
+
+    def get_class_attributes(self):
+        return {
+            "base": self.base,
+            "precision_fractional": self.precision_fractional,
+            # Solving coefficients can be reused in results of operations (see docstring of the reconstruct method)
+            "reconstruction_coeffs": self.reconstruction_coeffs,
+        }
+
+
+_moduli_for_fields = {
+    21: [3, 7],  # Still here for some small tests
+    "int64": [257, 263, 269, 271, 277, 281, 283, 293],
+    "int100": [1201, 1433, 1217, 1237, 1321, 1103, 1129, 1367, 1093, 1039],
```

Comment:
what is the size of the underlying types holding the values? int32? int64?

Reply:
FixedPrecisionTensors are used to hold the residue and they always are int64, I think.
I indeed think that it would make more sense to use int32 tensors behind CRTTensors, which means choosing the type of tensor when creating a FPT